src/test/java/ObjectTest.java

```diff
@@ -454,26 +454,22 @@ public void testObjectCreateBadAuthorizationNone() {
 	@Test(description = "object create w/negative content length, fails")
 	public void testObjectCreateBadContentlengthNegative() {
 
-		try {
-			String bucket_name = utils.getBucketName(prefix);
-			String key = "key1";
-			String content = "echo lima golf";
-			long contlength = -1;
+    String bucket_name = utils.getBucketName(prefix);
+    String key = "key1";
+    String content = "echo lima golf";
+    long contlength = -1;
 
-			svc.createBucket(new CreateBucketRequest(bucket_name));
+    svc.createBucket(new CreateBucketRequest(bucket_name));
 
-			byte[] contentBytes = content.getBytes(StringUtils.UTF8);
-			InputStream is = new ByteArrayInputStream(contentBytes);
+    byte[] contentBytes = content.getBytes(StringUtils.UTF8);
+    InputStream is = new ByteArrayInputStream(contentBytes);
 
-			ObjectMetadata metadata = new ObjectMetadata();
-			metadata.setHeader("Content-Length", contlength);
+    ObjectMetadata metadata = new ObjectMetadata();
+    metadata.setHeader("Content-Length", contlength);
 
-			svc.putObject(new PutObjectRequest(bucket_name, key, is, metadata));
-			AssertJUnit.fail("Expected 411 MissingContentLength");
-		} catch (AmazonServiceException err) {
-			AssertJUnit.assertNotSame(err.getLocalizedMessage(), null);
-			AssertJUnit.assertEquals(err.getErrorCode(), "MissingContentLength");
-		}
+    svc.putObject(new PutObjectRequest(bucket_name, key, is, metadata));
+    // following Ceph PR#50235 -- handle old clients with transfer-encoding: chunked
+    // this test PASS condition is 200 (versus 411 previously)
```

Comment:
```java
	@Test(description = "object create w/negative content length, fails")
	public void testObjectCreateBadContentlengthNegative() {
```
can you tell whether the client is sending that `Content-Length` header at all? it seems like rgw should return a 400 error there. maybe we're parsing the `-1`, but misinterpreting it as:
```c++
rgw_rest.h:static constexpr int64_t NO_CONTENT_LENGTH = -1;
```

Reply:
capturing with wireshark, the client (User-Agent: aws-sdk-java/1.11.549) is not sending a `Content-Length` header:

![image](https://user-images.githubusercontent.com/31659604/226373616-6ca45d72-3d8f-4f59-ad3f-c19c35bafccf.png)

Reply:
ok, thanks. since this test isn't even testing what it's supposed to, maybe we should remove it? or at least rename it to something like `testObjectCreateChunked`?

Reply:
sounds good